test/ETHStaticAToken.test.ts

```diff
@@ -1,126 +1,232 @@
         block: 2,
         acc_rewards_per_token: {
           high: 0n,
           low: 100n,
         },
       });
       expect.fail("non-owner was able to update rewards");
-    } catch (e) {
-    }
+    } catch (e) {}
   });
 
-  it('only allows increases in accRewards', async () => {
+  it("only allows increases in accRewards", async () => {
     try {
-      await owner.call(tokenContract, 'push_acc_rewards_per_token', {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
         block: 3,
         acc_rewards_per_token: {
           high: 0,
           low: 0,
-        }
+        },
       });
       expect.fail("accRewards was decreased");
-    } catch (e) {
-    }
+    } catch (e) {}
   });
 
-  it('rejects old block numbers', async () => {
+  it("rejects old block numbers", async () => {
     try {
-      await owner.call(tokenContract, 'push_acc_rewards_per_token', {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
         block: 0,
         acc_rewards_per_token: {
           high: 0,
           low: 2,
-        }
+        },
       });
       expect.fail("accRewards accepted for old block number");
-    } catch (e) {
-    }
+    } catch (e) {}
   });
 
-  it('updatesUserAccRewardsPerToken', async () => {
-    expect.fail("not implemented yet")
+  it("updatesUserAccRewardsPerToken", async () => {
+    try {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
+        block: 0,
+        acc_rewards_per_token: {
+          high: 0,
+          low: 2,
+        },
+      });
+      expect.fail("accRewards accepted for old block number");
+    } catch (e) {}
+  });
+
+  it("claims rewards to user", async () => {
+    try {
+      await user1.call(tokenContract, "claim_rewards", {
+        user: user1.starknetContract.address,
+        recipient: user1.starknetContract.address,
+      });
+
+      const { userPendingRewards } = await tokenContract.call(
+        "get_user_pending_rewards",
+        {
+          account: BigInt(user1.starknetContract.address),
+        }
+      );
+      const { userRewardsBalance } = await rewAaveTokenL2.call("balanceOf", {
+        account: BigInt(user1.starknetContract.address),
+      });
+
+      expect(userRewardsBalance).to.equal(userPendingRewards);
+    } catch (e) {}
   });
 
-  it('distributes rewards correctly', async () => {
-    expect.fail("not implemented yet")
+  it("updates acc rewards and checks for the exact value on the user", async () => {
+    try {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
+        block: 4,
+        acc_rewards_per_token: {
+          high: 0,
+          low: 3,
+        },
+      });
+
+      const { userAccruedRewards } = await tokenContract.call(
+        "get_user_acc_rewards_per_token",
+        {
+          account: BigInt(user1.starknetContract.address),
+        }
+      );
+      expect(userAccruedRewards).to.equal({
+        high: 0,
+        low: 3,
```

Comment:
Since the user hasn't interacted with the contract since the last update, at which point the accrued rewards per token was 2, shouldn't this be 2?

Reply:
You are right, that's weird it passed the test though

Reply:
There's no claim between `push_acc_rewards_per_token` and the `get_user_acc_rewards_per_token`